integrationTests/testConsensusNode.go

```diff
@@ -0,0 +1,462 @@
+		PrevRandSeed:  rootHash,
+		RandSeed:      rootHash,
+	}
+
+	_ = tcn.ChainHandler.SetGenesisHeader(header)
+	hdrMarshalized, _ := TestMarshalizer.Marshal(header)
+	tcn.ChainHandler.SetGenesisHeaderHash(hasher.Compute(string(hdrMarshalized)))
+}
+
+func (tcn *TestConsensusNode) initBlockProcessor() {
+	tcn.BlockProcessor = &mock.BlockProcessorMock{
+		ProcessBlockCalled: func(header data.HeaderHandler, body data.BodyHandler, haveTime func() time.Duration) error {
+			_ = tcn.ChainHandler.SetCurrentBlockHeaderAndRootHash(header, header.GetRootHash())
+			return nil
+		},
+		RevertCurrentBlockCalled: func() {
+		},
+		CreateBlockCalled: func(header data.HeaderHandler, haveTime func() bool) (data.HeaderHandler, data.BodyHandler, error) {
+			return header, &dataBlock.Body{}, nil
+		},
+		MarshalizedDataToBroadcastCalled: func(header data.HeaderHandler, body data.BodyHandler) (map[uint32][]byte, map[string][][]byte, error) {
+			mrsData := make(map[uint32][]byte)
+			mrsTxs := make(map[string][][]byte)
+			return mrsData, mrsTxs, nil
+		},
+		CreateNewHeaderCalled: func(round uint64, nonce uint64) (data.HeaderHandler, error) {
+			return &dataBlock.Header{
+				Round:           round,
+				Nonce:           nonce,
+				SoftwareVersion: []byte("version"),
+			}, nil
+		},
+	}
+
+	tcn.BlockProcessor.CommitBlockCalled = func(header data.HeaderHandler, body data.BodyHandler) error {
+		tcn.BlockProcessor.NrCommitBlockCalled++
+		_ = tcn.ChainHandler.SetCurrentBlockHeaderAndRootHash(header, header.GetRootHash())
+		return nil
+	}
+	tcn.BlockProcessor.Marshalizer = TestMarshalizer
+}
+
+func (tcn *TestConsensusNode) initResolverFinder() {
+	hdrResolver := &mock.HeaderResolverStub{}
+	mbResolver := &mock.MiniBlocksResolverStub{}
+	tcn.ResolverFinder = &mock.ResolversFinderStub{
+		IntraShardResolverCalled: func(baseTopic string) (resolver dataRetriever.Resolver, e error) {
+			if baseTopic == factory.MiniBlocksTopic {
+				return mbResolver, nil
+			}
+			return nil, nil
+		},
+		CrossShardResolverCalled: func(baseTopic string, crossShard uint32) (resolver dataRetriever.Resolver, err error) {
+			if baseTopic == factory.ShardBlocksTopic {
+				return hdrResolver, nil
+			}
+			return nil, nil
+		},
+	}
+}
+
+func (tcn *TestConsensusNode) initAccountsDB(hasher hashing.Hasher) {
+	// TODO change this implementation with a factory
+	generalCfg := config.TrieStorageManagerConfig{
+		PruningBufferLen:      1000,
+		SnapshotsBufferLen:    10,
+		SnapshotsGoroutineNum: 1,
+	}
+	args := trie.NewTrieStorageManagerArgs{
+		MainStorer:             CreateMemUnit(),
+		CheckpointsStorer:      CreateMemUnit(),
+		Marshalizer:            TestMarshaller,
+		Hasher:                 hasher,
+		GeneralConfig:          generalCfg,
+		CheckpointHashesHolder: hashesHolder.NewCheckpointHashesHolder(10000000, uint64(hasher.Size())),
+		IdleProvider:           &testscommon.ProcessStatusHandlerStub{},
+	}
+	trieStorage, _ := trie.NewTrieStorageManager(args)
```

Comment:
here i think we can refactor this to use the `CreateTrieStorageManager` function:
```suggestion
	storer := CreateMemUnit()
	trieStorage, _ := CreateTrieStorageManager(storer)
```

Reply:
i think here we can refactor to use `CreateTrieStorageManager` function